nft-barter-core/interfaces/INFTBarter.sol

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.15;
+
+/**
+@title an interface for swaping nfts
+@author Asim Mehmood
+@notice the contract enable users to swap their nft any other nft
+@dev It descriebe the signature to accept swaps given its swap id  and list all the swaps for token id
+ */
+abstract contract INFTBarter {
+    /**
+    @title swap order 
+    @notice A struct containing information needed for a swap
+     */
+    struct SwapOrder{
+        address makerAddress;
+        address takerAddress;
+        int152 valueDifference;
```

Comment:
As @fariedt pointed out, there should be a maximum value for this.  We should look at other tradings to see how they handle it.